components/butterfly/src/server/push.rs

```diff
@@ -38,7 +37,7 @@ lazy_static! {
 
 pub fn spawn_thread(name: String, server: Server, timing: Timing) -> std::io::Result<()> {
     thread::Builder::new().name(name)
-                          .spawn(move || run_loop(&server, &timing))
+                          .spawn(move || -> ! { run_loop(&server, &timing) })
```

Comment:
I notice that removing the `-> !` doesn't seem to have an effect, as long as `run_loop` is defined to return either nothing, or `!`.

As soon as I modified `run_loop` to return something, though, this refuses to compile. Neat!